python/cuml/fil/fil.pyx

```diff
@@ -478,6 +480,63 @@ class ForestInference(Base,
 
     """
 
+    def common_load_params_docstring(func):
+        func.__doc__ = getdoc(func).format("""
+    output_class: boolean (default=False)
+        For a Classification model `output_class` must be True.
+        For a Regression model `output_class` must be False.
+    algo : string (default='auto')
+        Name of the algo from (from algo_t enum):
+
+         - ``'AUTO'`` or ``'auto'``: Choose the algorithm automatically.
+           Currently 'BATCH_TREE_REORG' is used for dense storage,
+           and 'NAIVE' for sparse storage
+         - ``'NAIVE'`` or ``'naive'``: Simple inference using shared memory
+         - ``'TREE_REORG'`` or ``'tree_reorg'``: Similar to naive but trees
+           rearranged to be more coalescing-friendly
+         - ``'BATCH_TREE_REORG'`` or ``'batch_tree_reorg'``: Similar to
+           TREE_REORG but predicting multiple rows per thread block
+
+    threshold : float (default=0.5)
+        Threshold is used to for classification. It is applied
+        only if ``output_class == True``, else it is ignored.
+    storage_type : string or boolean (default='auto')
+        In-memory storage format to be used for the FIL model:
+
+         - ``'auto'``: Choose the storage type automatically
+           (currently DENSE is always used)
+         - ``False``: Create a dense forest
+         - ``True``: Create a sparse forest. Requires algo='NAIVE' or
+           algo='AUTO'
+
+    blocks_per_sm : integer (default=0)
+        (experimental) Indicates how the number of thread blocks to lauch
+        for the inference kernel is determined.
+
+        - ``0`` (default): Launches the number of blocks proportional to
+          the number of data rows
+        - ``>= 1``: Attempts to lauch blocks_per_sm blocks per SM. This
+          will fail if blocks_per_sm blocks result in more threads than the
+          maximum supported number of threads per GPU. Even if successful,
+          it is not guaranteed that blocks_per_sm blocks will run on an SM
+          concurrently.
+    compute_shape_str : boolean (default=False)
+        if True or equivalent, creates a ForestInference.shape_str
+        (writes a human-readable forest shape description as a
+        multiline ascii string)
+    """)
+        return func
+
+    def common_predict_params_docstring(func):
```

Comment:
I think this is repeating functionality available here: https://github.com/rapidsai/cuml/blob/f71d369da3a11cc498a551b712fae28f8a64f097/python/cuml/common/doc_utils.py#L352 

Was wondering if you tried using that existing decorator and if it caused any issues?

Reply:
I did, and (at least in cython), combining the docstring decorator with `@staticmethod` erases the effect of the docstring decorator. It is easily reproducible, and I did not find an answer to why it's happening.